Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -106,6 +119,55 @@ protected override BoolResult InitializeCore(OperationContext context)
             return result;
         }
 
+        /// <inheritdoc />
+        public override void SetDatabaseMode(bool isDatabaseWriteable)
+        {
+            if (IsGarbageCollectionEnabled != isDatabaseWriteable)
+            {
+                var nextCompactionTimeSpan = isDatabaseWriteable ? _configuration.FullRangeCompactionInterval : Timeout.InfiniteTimeSpan;
+                _compactionTimer?.Change(nextCompactionTimeSpan, Timeout.InfiniteTimeSpan);
+            }
+
+            base.SetDatabaseMode(isDatabaseWriteable);
+        }
+
+        private void FullRangeCompaction(OperationContext context)

```

Comment:
Need to trace this.

We need to know the duration of such a critical operation like compaction.

Reply:
This is being traced below, when effectively compacting each one of the column families.

---
In reply to: [313643384](https://github.com/microsoft/BuildXL/pull/731#discussion_r313643384) [](ancestors = 313643384)

Reply:
Nope. It wasn't. It was traced per family. We need overall time.

---
In reply to: [313645611](https://github.com/microsoft/BuildXL/pull/731#discussion_r313645611) [](ancestors = 313645611,313643384)

Reply:
I don't think it is that important to know per family. But I rather have one global time.

---
In reply to: [313645851](https://github.com/microsoft/BuildXL/pull/731#discussion_r313645851) [](ancestors = 313645851,313645611,313643384)

Reply:
Overall time is meaningless for this operation, as compaction time varies widely between each of the column families

---
In reply to: [313645851](https://github.com/microsoft/BuildXL/pull/731#discussion_r313645851) [](ancestors = 313645851,313645611,313643384)

Reply:
why it meaningless? For casaas with specific configuration it would be roughtly the same. No?

---
In reply to: [313646240](https://github.com/microsoft/BuildXL/pull/731#discussion_r313646240) [](ancestors = 313646240,313645851,313645611,313643384)